### PR TITLE
Add reply functionality, tests Feature/850

### PR DESCRIPTION
**Pull request by YuriiDubnytskyi**
Add reply functionality 
Add reply list
Add filters to comment list and reply list
Add new page to edit reply
Refactor comments list on product page

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅ All tests passed locally
- [x] ✨ My changes working with up-to-date front-end and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: src/components/forms/comment-form/comment-form.js [hunk 1]
```diff
@@ -68,8 +65,29 @@ const CommentForm = ({ comment, id, isEdit }) => {
...
+  };
+
   function handleProductClick() {
-    history.push(pathToEditProduct.replace(':id', comment.product._id));
+    if (comment.product?._id) {
+      return history.push(
+        pathToEditProduct.replace(':id', comment.product._id)
+      );
+    }
+    dispatch(showErrorSnackbar('Product not exist'));
```
Comment (Elugormo) [inline src/components/forms/comment-form/comment-form.js:90]: doesn't!!!

File: src/components/forms/reply-comment-form/reply-comment-form.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            showReplyComment: data.showReplyComment
+          },
+          commentId: commentIdUpdate
+        })
+      );
+    };
+    openSuccessSnackbar(updateReplyForComment, SAVE_MESSAGE, SAVE_CHANGES);
+  };
+
+  function handleCommentClick() {
```
Comment (Elugormo) [inline src/components/forms/reply-comment-form/reply-comment-form.js:105]: write functions in one style only, like you should choose either regular functions with "function" syntax or far arrow functions with () => {} syntax

**Review by Elugormo — CHANGES_REQUESTED** [review]

File: src/components/forms/reply-comment-form/reply-comment-form.js (hunk 2 above)
Comment (YuriiDubnytskyi) [reply src/components/forms/reply-comment-form/reply-comment-form.js]: fix

File: src/components/forms/comment-form/comment-form.js (hunk 1 above)
Comment (YuriiDubnytskyi) [reply src/components/forms/comment-form/comment-form.js]: fix

**Review by Elugormo — APPROVED** [review]

**Review by mvmoshchovskyi — APPROVED** [review]

**Review by Serhii-Ahafonov — APPROVED** [review]

**Pull request merged**